lib/CMakeLists.txt

```diff
@@ -1,4 +1,5 @@
 add_subdirectory(crc)
+add_subdirectory(quad_math)
```

Comment:
Need more detail on why this quad math library is needed.

What specifically is "broken" about clang's libm? I would think that this would be a dealbreaker for a lot of people.

Also I believe we have rules about importing third-party code into the tree which need to be followed, such as attribution in the commit message and a vanilla import of sources in its own commit.

Reply:
> Need more detail on why this quad math library is needed.

zephyr relies on libgcc for 64bit arithmetic and we have a few places  that depend on this, sensors, the hpet timer for example.
llvm does support this in some way, but it seems like it might be broken when doing cross-compilation, this comes with the compiler-rt library. 

> What specifically is "broken" about clang's libm?

I did not mention libm being broken anyways.

> Also I believe we have rules about importing third-party code into the tree which need to be followed, such as attribution in the commit message and a vanilla import of sources in its own commit.

I was hoping I would not need this if I could find a better solution, so I did not get into the official part of importing this code.